xbmc/dbwrappers/DatabaseQuery.cpp

```diff
@@ -456,8 +462,7 @@ std::string CDatabaseQueryRuleCombination::GetWhereClause(const CDatabase& db,
   std::string rule;
 
   // translate the combinations into SQL
-  for (CDatabaseQueryRuleCombinations::const_iterator it = m_combinations.begin();
-       it != m_combinations.end(); ++it)
+  for (auto it = m_combinations.begin(); it != m_combinations.end(); ++it)
```

Comment:
```suggestion
  for (const auto &it : m_combinations)
```

Reply:
Also requires further changes. See some lines below the for statement

`if (it + 1 != m_parameter.end())`

I don't want to tackle this in this PR. I really want to concentrate on the SQ findings. There is so much more in this (ancient) source files that could be improved...